api/src/main/java/com/graphhopper/config/ProfileConfig.java

```diff
@@ -0,0 +1,108 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.graphhopper.config;
+
+import com.graphhopper.util.PMap;
+
+public class ProfileConfig {
+    private String name = "car";
+    private String vehicle = "car";
+    private String weighting = "fastest";
+    private boolean turnCosts = false;
+    private PMap hints = new PMap();
+
+    public static void validateProfileName(String profileName) {
+        if (!profileName.matches("^[a-z0-9_\\-]*$")) {
+            throw new IllegalArgumentException("Profile names may only contain lower case letters, numbers, underscores and dashs, given: " + profileName);
+        }
+    }
+
+    private ProfileConfig() {
+        // default constructor needed for jackson
+    }
+
+    public ProfileConfig(String name) {
+        setName(name);
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public ProfileConfig setName(String name) {
+        validateProfileName(name);
+        this.name = name;
+        return this;
+    }
+
+    public String getVehicle() {
+        return vehicle;
+    }
+
+    public ProfileConfig setVehicle(String vehicle) {
+        this.vehicle = vehicle;
+        return this;
+    }
+
+    public String getWeighting() {
+        return weighting;
+    }
+
+    public ProfileConfig setWeighting(String weighting) {
+        this.weighting = weighting;
+        return this;
+    }
+
+    public boolean isTurnCosts() {
+        return turnCosts;
+    }
+
+    public ProfileConfig setTurnCosts(boolean turnCosts) {
+        this.turnCosts = turnCosts;
+        return this;
+    }
+
+    public PMap hints() {
```

Comment:
Why is it not named `getHints`?

Reply:
Because of the jackson conventions. If there was getHints() Jackson would allow a hints sub-node, but we want all the other fields on the top level (see the `@JsonAnySetter` in the corresponding mixin)

Reply:
What do you mean with "would allow a hints sub-node" - when would it allow this? As there is no setHints method.

Reply:
The getter makes the hints serializable and deserializable: https://www.baeldung.com/jackson-field-serializable-deserializable-or-not#a-getter-makes-a-non-public-field-serializable-and-deserializable. So if we put `getHints` by convention the deserialized config would look like:
```yaml
profile:
  name: my_name
  hints: 
      foo: bar
      baz: foo
```

but we want:

```yaml
profile:
  name: my_name
  foo: bar
  baz: foo
```

Reply:
And can we use JsonIgnore?

Reply:
Yes with `@JsonIgnore` we can have `getHints` and Jackson will still not consider `hints` to be a property. -> I renamed to `getHints()`.